Frontend/ui-customer/src/Flow.purs

```diff
@@ -937,19 +937,36 @@ homeScreenFlow = do
-                _ <- setValueToLocalStore AUTO_SELECTING "false"
-                _ <- pure $ updateLocalStage QuoteList
-                modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = QuoteList,selectedQuote = Nothing, expiredQuotes = snoc state.props.expiredQuotes (fromMaybe "" state.props.selectedQuote)}, data {quoteListModelState = []}})
-                homeScreenFlow
-            else homeScreenFlow
+            let isBusQuoteSelected = state.props.isBusQuoteSelected
+                quantity = state.props.quantity
+            if not isBusQuoteSelected then do
+              response  <- lift $ lift $ Remote.rideConfirm (fromMaybe "" selectedQuote)
+              case response of
+                Right (ConfirmRes resp) -> do
+                  let bookingId = resp.bookingId
+                  modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = ConfirmingRide, bookingId = bookingId, isPopUp = NoPopUp}})
+                  homeScreenFlow
+                Left err  -> do
+                  if not (err.code == 400 && (decodeError err.response.errorMessage "errorCode") == "QUOTE_EXPIRED") then pure $ toast (getString ERROR_OCCURED_TRY_AGAIN) else pure unit
+                  _ <- setValueToLocalStore AUTO_SELECTING "false"
+                  _ <- pure $ updateLocalStage QuoteList
+                  modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = QuoteList,selectedQuote = Nothing, expiredQuotes = snoc state.props.expiredQuotes (fromMaybe "" state.props.selectedQuote)}, data {quoteListModelState = []}})
+                  homeScreenFlow
+            else do
+              _ <- pure $ spy "selected Quote " selectedQuote
+              response  <- lift $ lift $ Remote.busConfirm (fromMaybe "" selectedQuote) quantity
+              case response of
+                Right (BusConfirmRes resp) -> do
+                  let ticketId = resp.ticketId
+                  modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = ConfirmingRide, ticketId = ticketId, isPopUp = NoPopUp}})
+                  homeScreenFlow
+                Left err  -> do
+                  if not (err.code == 400 && (decodeError err.response.errorMessage "errorCode") == "QUOTE_EXPIRED") then pure $ toast (getString ERROR_OCCURED_TRY_AGAIN) else pure unit
+                  _ <- setValueToLocalStore AUTO_SELECTING "false"
+                  _ <- pure $ updateLocalStage QuoteList
+                  modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen {props{currentStage = QuoteList, selectedQuote = Nothing, expiredQuotes = snoc state.props.expiredQuotes (fromMaybe "" state.props.selectedQuote)}, data {quoteListModelState = []}})
+                  homeScreenFlow
+          else homeScreenFlow
```

Comment:
can you create common error function for this

Reply:
removed from `ui-customer/ui-driver` and moved to `ui-common`